Automattic/jetpack
# Jetpack Backup: Refactor component into multiple modules

Fixes #14065 and prepares the Plans codebase for Jetpack Search.

#### Changes proposed in this Pull Request:
* Modularizes several components within `single-product-backup.jsx` into reusable components. 
* Use `useMemo` for generating `backupOptions` object.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No, this is a feature-complete refactor of the Plans page for Jetpack Backups.

#### Testing instructions:
1. Setup this branch of Jetpack on your site (or [use Docker](https://github.com/Automattic/jetpack/tree/master/docker#readme)).
2. Ensure that you have no Jetpack plan associated with your site. Navigate to `/wp-admin/admin.php?page=jetpack#/plans` and scroll down to the Solutions section. You should see something like the following:

> No Plans: 
>
> <img width="546" alt="Screen Shot 2020-03-11 at 5 50 21 PM" src="https://user-images.githubusercontent.com/4044428/76474388-217e8880-63c1-11ea-9ce0-d7cff4ec3491.png">

> Personal Plan:
>
> <img width="546" alt="Screen Shot 2020-03-11 at 5 50 30 PM" src="https://user-images.githubusercontent.com/4044428/76474380-1af01100-63c1-11ea-9207-c36b08b8ff99.png">

> Real-time Backup Plan:
>
> <img width="546" alt="Screen Shot 2020-03-11 at 5 50 39 PM" src="https://user-images.githubusercontent.com/4044428/76474375-14fa3000-63c1-11ea-9dc8-05a24e260ced.png">

> Professional Plan:
>
> <img width="546" alt="Screen Shot 2020-03-11 at 5 50 45 PM" src="https://user-images.githubusercontent.com/4044428/76474363-0d3a8b80-63c1-11ea-9822-ef98486a5d45.png">

#### Proposed changelog entry for your changes:

* No changelog necessary.
